projects/observability/src/shared/components/span-detail/span-detail.component.ts

```diff
@@ -79,12 +99,39 @@ export class SpanDetailComponent implements OnChanges {
   @Output()
   public readonly closed: EventEmitter<void> = new EventEmitter<void>();
 
+  @ViewChild('requestTemplate', { static: true })
+  public requestTemplate!: TemplateRef<unknown>;
+
+  @ViewChild('responseTemplate', { static: true })
+  public responseTemplate!: TemplateRef<unknown>;
+
+  @ViewChild('attributesTemplate', { static: true })
+  public attributesTemplate!: TemplateRef<unknown>;
+
+  @ViewChild('exitCallsTemplate', { static: true })
+  public exitCallsTemplate!: TemplateRef<unknown>;
+
+  @ViewChild('logEventsTemplate', { static: true })
+  public logEventsTemplate!: TemplateRef<unknown>;
+
   public showRequestTab?: boolean;
   public showResponseTab?: boolean;
   public showExitCallsTab?: boolean;
   public showLogEventsTab?: boolean;
   public totalLogEvents?: number;
 
+  public tabs: ToggleItem<SpanDetailTab>[] = [];
+
+  private readonly activeTabSubject: BehaviorSubject<ToggleItem<SpanDetailTab> | undefined> = new BehaviorSubject<
+    ToggleItem<SpanDetailTab> | undefined
+  >(undefined);
+  public readonly activeTab$: Observable<ToggleItem<SpanDetailTab> | undefined> = this.activeTabSubject.asObservable();
+
+  private readonly templateSubject: BehaviorSubject<TemplateRef<unknown> | undefined> = new BehaviorSubject<
+    TemplateRef<unknown> | undefined
+  >(undefined);
+  public readonly template$: Observable<TemplateRef<unknown> | undefined> = this.templateSubject.asObservable();
```

Comment:
This seems weird to me. Why not just do ngSwitch on SpanDetailTab in the template?

Reply:
Will refactor to use `ngSwitch` instead.

Reply:
Updated!